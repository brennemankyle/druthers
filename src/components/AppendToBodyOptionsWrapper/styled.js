import styled from "@emotion/styled";
import { StyledAppendToBodyOptionsWrapper as RawStyledAppendToBodyOptionsWrapper } from "./AppendToBodyOptionsWrapper";

const StyledAppendToBodyOptionsWrapper = styled(
  RawStyledAppendToBodyOptionsWrapper
)`
  position: absolute;
  background-color: ${props => props.styles_colors_background};
  width: ${props => props.styles_width}px;

  ${props =>
    props.placeOptionsAbove
      ? `bottom: ${window.innerHeight - props.parentRect.y}px;
      left: ${props.parentRect.x}px;`
      : `top: ${props.parentRect.y + props.parentRect.height}px;
      left: ${props.parentRect.x}px;`}
  ${props => props.styles_rightToLeft && `transform: scaleX(-1);`}
  ${props =>
    !props.styles_hasOptions && !props.styles_multiple && `display: none;`}
`;

export { StyledAppendToBodyOptionsWrapper };
