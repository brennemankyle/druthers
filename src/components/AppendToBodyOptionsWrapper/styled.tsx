import React from "react";
import styled from "@emotion/styled";
import {
  StyledAppendToBodyOptionsWrapper as RawStyledAppendToBodyOptionsWrapper,
  StyledAppendToBodyOptionsWrapperProps,
} from "./AppendToBodyOptionsWrapper";
import { Styles } from "../../utils/SelectTypes";
import { createBaseOptionsWrapper } from "../Wrapper/styled";

const BaseOptionsWrapper = createBaseOptionsWrapper(
  RawStyledAppendToBodyOptionsWrapper
);

const StyledAppendToBodyOptionsWrapper = styled(BaseOptionsWrapper)(
  (props: Styles & StyledAppendToBodyOptionsWrapperProps) => `
  position: absolute;
  background-color: ${props.styles_colors_background};

  ${
    props.placeOptionsAbove
      ? `bottom: ${window.innerHeight - props.parentRect.y}px;
        left: ${props.parentRect.x}px;`
      : `top: ${props.parentRect.y + props.parentRect.height}px;
        left: ${props.parentRect.x}px;`
  }
  ${props.styles_rightToLeft ? `transform: scaleX(-1);` : ``}
  ${!props.styles_hasOptions && !props.styles_multiple ? `display: none;` : ``}
`
);

export { StyledAppendToBodyOptionsWrapper };
