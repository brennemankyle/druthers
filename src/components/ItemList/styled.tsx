import React from "react";
import styled from "@emotion/styled";
import ItemList from "./ItemList";

const SelectionList = styled(ItemList)(
  (props) => `
  ${
    props.styles_multiple
      ? `display: inline;`
      : `display: inline-block;
      &:not(:empty) {
        width: 100%;
      }`
  }
  list-style-type: none;
  padding: 0;
  margin: 0;
  color: ${props.styles_colors_text};
`
);

const OptionList = styled(ItemList)(
  (props) => `
  color: ${props.styles_colors_text};
  list-style-type: none;
  padding: 0;
  margin: 0;
  margin-block-start: 0;
  margin-block-end: 0;
  border-color: ${props.styles_colors_secondary};
  border-width: ${props.styles_borderWidth};
  border-style: ${props.styles_borderStyle};
  border-radius: ${props.styles_borderRadius};

  .truncate-show, .truncate-hide {
    cursor: pointer;
  }
`
);

export { SelectionList, OptionList };
