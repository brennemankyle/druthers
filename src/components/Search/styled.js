import styled from "@emotion/styled";
import RawSearch from "./Search";

const Search = styled(RawSearch)`
  border: none;
  outline: 0;
  font-size: 1em;
  font-family: inherit;
  background-color: transparent;
  &::-ms-clear {
    display: none;
  }
  margin-top: calc(${props => props.styles_selection_margin} / 2);
  margin-bottom: calc(${props => props.styles_selection_margin} / 2);
  padding-top: ${props => props.styles_selection_paddingTop};
  padding-bottom: ${props => props.styles_selection_paddingBottom};
  padding-right: ${props => props.styles_selection_paddingRight};
  padding-left: ${props => props.styles_selection_paddingLeft};
  ${props =>
    props.styles_rightToLeft &&
    `transform: scaleX(-1);
    text-align: right;`}
  ${props =>
    props.hide &&
    `opacity: 0;
    width: 0;
    padding: 0;
    margin: 0;`}
  ${props => props.styles_disabled && `cursor: not-allowed;`}
`;

export { Search };
