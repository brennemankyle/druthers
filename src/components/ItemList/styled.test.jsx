import React from "react";
import renderer from "react-test-renderer";
import { mockStyles, mockItemList, MockElement } from "../../mocks";
import { SelectionList, OptionList } from "./styled";

describe("SelectionList", () => {
  it("styles", () => {
    const wrapper = renderer.create(
      <SelectionList
        {...mockStyles}
        itemList={mockItemList}
        Item={MockElement}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("styles_multiple = true", () => {
    const wrapper = renderer.create(
      <SelectionList
        {...mockStyles}
        itemList={mockItemList}
        Item={MockElement}
        styles_multiple={true}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});

describe("OptionList", () => {
  it("styles", () => {
    const wrapper = renderer.create(
      <OptionList {...mockStyles} itemList={mockItemList} Item={MockElement} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
