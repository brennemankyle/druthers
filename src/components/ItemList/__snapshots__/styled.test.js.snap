// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionList styles 1`] = `
.emotion-0 {
  list-style-type: none;
  padding: 0;
  margin: 0;
  margin-block-start: 0;
  margin-block-end: 0;
  border: 1px solid #C3C3C3;
  border-radius: .2em;
}

<ul
  className="emotion-0"
  onMouseDown={[Function]}
  onTouchStart={[Function]}
>
  <div
    data-props={
      Object {
        "item": Object {
          "label": "Item 1",
          "value": "1",
        },
        "removable": false,
        "styles_borderRadius": ".2em",
        "styles_checkRadio_borderWidth": "2px",
        "styles_checkRadio_labelMargin": ".2em",
        "styles_checkRadio_marginBetween": "1em",
        "styles_checkRadio_paddingBottom": ".2em",
        "styles_checkRadio_paddingLeft": ".2em",
        "styles_checkRadio_paddingRight": ".2em",
        "styles_checkRadio_paddingTop": ".2em",
        "styles_colors_background": "white",
        "styles_colors_disabled": "#ECECEC",
        "styles_colors_highlight": "lightblue",
        "styles_colors_primary": "black",
        "styles_colors_secondary": "#C3C3C3",
        "styles_colors_warning": "#FABAAC",
        "styles_colors_warningBold": "#FA2222",
        "styles_disabled": false,
        "styles_fontSize": "1em",
        "styles_hasOptions": true,
        "styles_hasSelection": false,
        "styles_icon_width": "1em",
        "styles_multiple": false,
        "styles_optionHighlighted": Array [],
        "styles_option_paddingBottom": ".25em",
        "styles_option_paddingLeft": ".25em",
        "styles_option_paddingRight": ".25em",
        "styles_option_paddingTop": ".25em",
        "styles_paddingBottom": ".25em",
        "styles_paddingLeft": ".25em",
        "styles_paddingRight": ".25em",
        "styles_paddingTop": ".25em",
        "styles_rightToLeft": false,
        "styles_search_size": "8",
        "styles_selection_margin": ".2em",
        "styles_selection_paddingBottom": ".2em",
        "styles_selection_paddingLeft": ".4em",
        "styles_selection_paddingRight": ".4em",
        "styles_selection_paddingTop": ".2em",
        "styles_width": 600,
        "svg_Remove": undefined,
      }
    }
  />
  <div
    data-props={
      Object {
        "item": Object {
          "label": "Item 2",
          "value": "2",
        },
        "removable": false,
        "styles_borderRadius": ".2em",
        "styles_checkRadio_borderWidth": "2px",
        "styles_checkRadio_labelMargin": ".2em",
        "styles_checkRadio_marginBetween": "1em",
        "styles_checkRadio_paddingBottom": ".2em",
        "styles_checkRadio_paddingLeft": ".2em",
        "styles_checkRadio_paddingRight": ".2em",
        "styles_checkRadio_paddingTop": ".2em",
        "styles_colors_background": "white",
        "styles_colors_disabled": "#ECECEC",
        "styles_colors_highlight": "lightblue",
        "styles_colors_primary": "black",
        "styles_colors_secondary": "#C3C3C3",
        "styles_colors_warning": "#FABAAC",
        "styles_colors_warningBold": "#FA2222",
        "styles_disabled": false,
        "styles_fontSize": "1em",
        "styles_hasOptions": true,
        "styles_hasSelection": false,
        "styles_icon_width": "1em",
        "styles_multiple": false,
        "styles_optionHighlighted": Array [],
        "styles_option_paddingBottom": ".25em",
        "styles_option_paddingLeft": ".25em",
        "styles_option_paddingRight": ".25em",
        "styles_option_paddingTop": ".25em",
        "styles_paddingBottom": ".25em",
        "styles_paddingLeft": ".25em",
        "styles_paddingRight": ".25em",
        "styles_paddingTop": ".25em",
        "styles_rightToLeft": false,
        "styles_search_size": "8",
        "styles_selection_margin": ".2em",
        "styles_selection_paddingBottom": ".2em",
        "styles_selection_paddingLeft": ".4em",
        "styles_selection_paddingRight": ".4em",
        "styles_selection_paddingTop": ".2em",
        "styles_width": 600,
        "svg_Remove": undefined,
      }
    }
  />
  <div
    data-props={
      Object {
        "item": Object {
          "label": "Item 3",
          "value": "3",
        },
        "removable": false,
        "styles_borderRadius": ".2em",
        "styles_checkRadio_borderWidth": "2px",
        "styles_checkRadio_labelMargin": ".2em",
        "styles_checkRadio_marginBetween": "1em",
        "styles_checkRadio_paddingBottom": ".2em",
        "styles_checkRadio_paddingLeft": ".2em",
        "styles_checkRadio_paddingRight": ".2em",
        "styles_checkRadio_paddingTop": ".2em",
        "styles_colors_background": "white",
        "styles_colors_disabled": "#ECECEC",
        "styles_colors_highlight": "lightblue",
        "styles_colors_primary": "black",
        "styles_colors_secondary": "#C3C3C3",
        "styles_colors_warning": "#FABAAC",
        "styles_colors_warningBold": "#FA2222",
        "styles_disabled": false,
        "styles_fontSize": "1em",
        "styles_hasOptions": true,
        "styles_hasSelection": false,
        "styles_icon_width": "1em",
        "styles_multiple": false,
        "styles_optionHighlighted": Array [],
        "styles_option_paddingBottom": ".25em",
        "styles_option_paddingLeft": ".25em",
        "styles_option_paddingRight": ".25em",
        "styles_option_paddingTop": ".25em",
        "styles_paddingBottom": ".25em",
        "styles_paddingLeft": ".25em",
        "styles_paddingRight": ".25em",
        "styles_paddingTop": ".25em",
        "styles_rightToLeft": false,
        "styles_search_size": "8",
        "styles_selection_margin": ".2em",
        "styles_selection_paddingBottom": ".2em",
        "styles_selection_paddingLeft": ".4em",
        "styles_selection_paddingRight": ".4em",
        "styles_selection_paddingTop": ".2em",
        "styles_width": 600,
        "svg_Remove": undefined,
      }
    }
  />
  <div
    data-props={
      Object {
        "item": Object {
          "label": "Item 4",
          "value": "4",
        },
        "removable": false,
        "styles_borderRadius": ".2em",
        "styles_checkRadio_borderWidth": "2px",
        "styles_checkRadio_labelMargin": ".2em",
        "styles_checkRadio_marginBetween": "1em",
        "styles_checkRadio_paddingBottom": ".2em",
        "styles_checkRadio_paddingLeft": ".2em",
        "styles_checkRadio_paddingRight": ".2em",
        "styles_checkRadio_paddingTop": ".2em",
        "styles_colors_background": "white",
        "styles_colors_disabled": "#ECECEC",
        "styles_colors_highlight": "lightblue",
        "styles_colors_primary": "black",
        "styles_colors_secondary": "#C3C3C3",
        "styles_colors_warning": "#FABAAC",
        "styles_colors_warningBold": "#FA2222",
        "styles_disabled": false,
        "styles_fontSize": "1em",
        "styles_hasOptions": true,
        "styles_hasSelection": false,
        "styles_icon_width": "1em",
        "styles_multiple": false,
        "styles_optionHighlighted": Array [],
        "styles_option_paddingBottom": ".25em",
        "styles_option_paddingLeft": ".25em",
        "styles_option_paddingRight": ".25em",
        "styles_option_paddingTop": ".25em",
        "styles_paddingBottom": ".25em",
        "styles_paddingLeft": ".25em",
        "styles_paddingRight": ".25em",
        "styles_paddingTop": ".25em",
        "styles_rightToLeft": false,
        "styles_search_size": "8",
        "styles_selection_margin": ".2em",
        "styles_selection_paddingBottom": ".2em",
        "styles_selection_paddingLeft": ".4em",
        "styles_selection_paddingRight": ".4em",
        "styles_selection_paddingTop": ".2em",
        "styles_width": 600,
        "svg_Remove": undefined,
      }
    }
  />
  <div
    data-props={
      Object {
        "item": Object {
          "label": "Item 5",
          "value": "5",
        },
        "removable": false,
        "styles_borderRadius": ".2em",
        "styles_checkRadio_borderWidth": "2px",
        "styles_checkRadio_labelMargin": ".2em",
        "styles_checkRadio_marginBetween": "1em",
        "styles_checkRadio_paddingBottom": ".2em",
        "styles_checkRadio_paddingLeft": ".2em",
        "styles_checkRadio_paddingRight": ".2em",
        "styles_checkRadio_paddingTop": ".2em",
        "styles_colors_background": "white",
        "styles_colors_disabled": "#ECECEC",
        "styles_colors_highlight": "lightblue",
        "styles_colors_primary": "black",
        "styles_colors_secondary": "#C3C3C3",
        "styles_colors_warning": "#FABAAC",
        "styles_colors_warningBold": "#FA2222",
        "styles_disabled": false,
        "styles_fontSize": "1em",
        "styles_hasOptions": true,
        "styles_hasSelection": false,
        "styles_icon_width": "1em",
        "styles_multiple": false,
        "styles_optionHighlighted": Array [],
        "styles_option_paddingBottom": ".25em",
        "styles_option_paddingLeft": ".25em",
        "styles_option_paddingRight": ".25em",
        "styles_option_paddingTop": ".25em",
        "styles_paddingBottom": ".25em",
        "styles_paddingLeft": ".25em",
        "styles_paddingRight": ".25em",
        "styles_paddingTop": ".25em",
        "styles_rightToLeft": false,
        "styles_search_size": "8",
        "styles_selection_margin": ".2em",
        "styles_selection_paddingBottom": ".2em",
        "styles_selection_paddingLeft": ".4em",
        "styles_selection_paddingRight": ".4em",
        "styles_selection_paddingTop": ".2em",
        "styles_width": 600,
        "svg_Remove": undefined,
      }
    }
  />
</ul>
`;

exports[`SelectionList styles 1`] = `
.emotion-0 {
  display: inline-block;
  list-style-type: none;
  padding: 0;
  margin: 0;
}

.emotion-0:not(:empty) {
  width: 100%;
}

<ul
  className="emotion-0"
  onMouseDown={[Function]}
  onTouchStart={[Function]}
>
  <div
    data-props={
      Object {
        "item": Object {
          "label": "Item 1",
          "value": "1",
        },
        "removable": false,
        "styles_borderRadius": ".2em",
        "styles_checkRadio_borderWidth": "2px",
        "styles_checkRadio_labelMargin": ".2em",
        "styles_checkRadio_marginBetween": "1em",
        "styles_checkRadio_paddingBottom": ".2em",
        "styles_checkRadio_paddingLeft": ".2em",
        "styles_checkRadio_paddingRight": ".2em",
        "styles_checkRadio_paddingTop": ".2em",
        "styles_colors_background": "white",
        "styles_colors_disabled": "#ECECEC",
        "styles_colors_highlight": "lightblue",
        "styles_colors_primary": "black",
        "styles_colors_secondary": "#C3C3C3",
        "styles_colors_warning": "#FABAAC",
        "styles_colors_warningBold": "#FA2222",
        "styles_disabled": false,
        "styles_fontSize": "1em",
        "styles_hasOptions": true,
        "styles_hasSelection": false,
        "styles_icon_width": "1em",
        "styles_multiple": false,
        "styles_optionHighlighted": Array [],
        "styles_option_paddingBottom": ".25em",
        "styles_option_paddingLeft": ".25em",
        "styles_option_paddingRight": ".25em",
        "styles_option_paddingTop": ".25em",
        "styles_paddingBottom": ".25em",
        "styles_paddingLeft": ".25em",
        "styles_paddingRight": ".25em",
        "styles_paddingTop": ".25em",
        "styles_rightToLeft": false,
        "styles_search_size": "8",
        "styles_selection_margin": ".2em",
        "styles_selection_paddingBottom": ".2em",
        "styles_selection_paddingLeft": ".4em",
        "styles_selection_paddingRight": ".4em",
        "styles_selection_paddingTop": ".2em",
        "styles_width": 600,
        "svg_Remove": undefined,
      }
    }
  />
  <div
    data-props={
      Object {
        "item": Object {
          "label": "Item 2",
          "value": "2",
        },
        "removable": false,
        "styles_borderRadius": ".2em",
        "styles_checkRadio_borderWidth": "2px",
        "styles_checkRadio_labelMargin": ".2em",
        "styles_checkRadio_marginBetween": "1em",
        "styles_checkRadio_paddingBottom": ".2em",
        "styles_checkRadio_paddingLeft": ".2em",
        "styles_checkRadio_paddingRight": ".2em",
        "styles_checkRadio_paddingTop": ".2em",
        "styles_colors_background": "white",
        "styles_colors_disabled": "#ECECEC",
        "styles_colors_highlight": "lightblue",
        "styles_colors_primary": "black",
        "styles_colors_secondary": "#C3C3C3",
        "styles_colors_warning": "#FABAAC",
        "styles_colors_warningBold": "#FA2222",
        "styles_disabled": false,
        "styles_fontSize": "1em",
        "styles_hasOptions": true,
        "styles_hasSelection": false,
        "styles_icon_width": "1em",
        "styles_multiple": false,
        "styles_optionHighlighted": Array [],
        "styles_option_paddingBottom": ".25em",
        "styles_option_paddingLeft": ".25em",
        "styles_option_paddingRight": ".25em",
        "styles_option_paddingTop": ".25em",
        "styles_paddingBottom": ".25em",
        "styles_paddingLeft": ".25em",
        "styles_paddingRight": ".25em",
        "styles_paddingTop": ".25em",
        "styles_rightToLeft": false,
        "styles_search_size": "8",
        "styles_selection_margin": ".2em",
        "styles_selection_paddingBottom": ".2em",
        "styles_selection_paddingLeft": ".4em",
        "styles_selection_paddingRight": ".4em",
        "styles_selection_paddingTop": ".2em",
        "styles_width": 600,
        "svg_Remove": undefined,
      }
    }
  />
  <div
    data-props={
      Object {
        "item": Object {
          "label": "Item 3",
          "value": "3",
        },
        "removable": false,
        "styles_borderRadius": ".2em",
        "styles_checkRadio_borderWidth": "2px",
        "styles_checkRadio_labelMargin": ".2em",
        "styles_checkRadio_marginBetween": "1em",
        "styles_checkRadio_paddingBottom": ".2em",
        "styles_checkRadio_paddingLeft": ".2em",
        "styles_checkRadio_paddingRight": ".2em",
        "styles_checkRadio_paddingTop": ".2em",
        "styles_colors_background": "white",
        "styles_colors_disabled": "#ECECEC",
        "styles_colors_highlight": "lightblue",
        "styles_colors_primary": "black",
        "styles_colors_secondary": "#C3C3C3",
        "styles_colors_warning": "#FABAAC",
        "styles_colors_warningBold": "#FA2222",
        "styles_disabled": false,
        "styles_fontSize": "1em",
        "styles_hasOptions": true,
        "styles_hasSelection": false,
        "styles_icon_width": "1em",
        "styles_multiple": false,
        "styles_optionHighlighted": Array [],
        "styles_option_paddingBottom": ".25em",
        "styles_option_paddingLeft": ".25em",
        "styles_option_paddingRight": ".25em",
        "styles_option_paddingTop": ".25em",
        "styles_paddingBottom": ".25em",
        "styles_paddingLeft": ".25em",
        "styles_paddingRight": ".25em",
        "styles_paddingTop": ".25em",
        "styles_rightToLeft": false,
        "styles_search_size": "8",
        "styles_selection_margin": ".2em",
        "styles_selection_paddingBottom": ".2em",
        "styles_selection_paddingLeft": ".4em",
        "styles_selection_paddingRight": ".4em",
        "styles_selection_paddingTop": ".2em",
        "styles_width": 600,
        "svg_Remove": undefined,
      }
    }
  />
  <div
    data-props={
      Object {
        "item": Object {
          "label": "Item 4",
          "value": "4",
        },
        "removable": false,
        "styles_borderRadius": ".2em",
        "styles_checkRadio_borderWidth": "2px",
        "styles_checkRadio_labelMargin": ".2em",
        "styles_checkRadio_marginBetween": "1em",
        "styles_checkRadio_paddingBottom": ".2em",
        "styles_checkRadio_paddingLeft": ".2em",
        "styles_checkRadio_paddingRight": ".2em",
        "styles_checkRadio_paddingTop": ".2em",
        "styles_colors_background": "white",
        "styles_colors_disabled": "#ECECEC",
        "styles_colors_highlight": "lightblue",
        "styles_colors_primary": "black",
        "styles_colors_secondary": "#C3C3C3",
        "styles_colors_warning": "#FABAAC",
        "styles_colors_warningBold": "#FA2222",
        "styles_disabled": false,
        "styles_fontSize": "1em",
        "styles_hasOptions": true,
        "styles_hasSelection": false,
        "styles_icon_width": "1em",
        "styles_multiple": false,
        "styles_optionHighlighted": Array [],
        "styles_option_paddingBottom": ".25em",
        "styles_option_paddingLeft": ".25em",
        "styles_option_paddingRight": ".25em",
        "styles_option_paddingTop": ".25em",
        "styles_paddingBottom": ".25em",
        "styles_paddingLeft": ".25em",
        "styles_paddingRight": ".25em",
        "styles_paddingTop": ".25em",
        "styles_rightToLeft": false,
        "styles_search_size": "8",
        "styles_selection_margin": ".2em",
        "styles_selection_paddingBottom": ".2em",
        "styles_selection_paddingLeft": ".4em",
        "styles_selection_paddingRight": ".4em",
        "styles_selection_paddingTop": ".2em",
        "styles_width": 600,
        "svg_Remove": undefined,
      }
    }
  />
  <div
    data-props={
      Object {
        "item": Object {
          "label": "Item 5",
          "value": "5",
        },
        "removable": false,
        "styles_borderRadius": ".2em",
        "styles_checkRadio_borderWidth": "2px",
        "styles_checkRadio_labelMargin": ".2em",
        "styles_checkRadio_marginBetween": "1em",
        "styles_checkRadio_paddingBottom": ".2em",
        "styles_checkRadio_paddingLeft": ".2em",
        "styles_checkRadio_paddingRight": ".2em",
        "styles_checkRadio_paddingTop": ".2em",
        "styles_colors_background": "white",
        "styles_colors_disabled": "#ECECEC",
        "styles_colors_highlight": "lightblue",
        "styles_colors_primary": "black",
        "styles_colors_secondary": "#C3C3C3",
        "styles_colors_warning": "#FABAAC",
        "styles_colors_warningBold": "#FA2222",
        "styles_disabled": false,
        "styles_fontSize": "1em",
        "styles_hasOptions": true,
        "styles_hasSelection": false,
        "styles_icon_width": "1em",
        "styles_multiple": false,
        "styles_optionHighlighted": Array [],
        "styles_option_paddingBottom": ".25em",
        "styles_option_paddingLeft": ".25em",
        "styles_option_paddingRight": ".25em",
        "styles_option_paddingTop": ".25em",
        "styles_paddingBottom": ".25em",
        "styles_paddingLeft": ".25em",
        "styles_paddingRight": ".25em",
        "styles_paddingTop": ".25em",
        "styles_rightToLeft": false,
        "styles_search_size": "8",
        "styles_selection_margin": ".2em",
        "styles_selection_paddingBottom": ".2em",
        "styles_selection_paddingLeft": ".4em",
        "styles_selection_paddingRight": ".4em",
        "styles_selection_paddingTop": ".2em",
        "styles_width": 600,
        "svg_Remove": undefined,
      }
    }
  />
</ul>
`;

exports[`SelectionList styles_multiple = true 1`] = `
.emotion-0 {
  display: inline;
  list-style-type: none;
  padding: 0;
  margin: 0;
}

<ul
  className="emotion-0"
  onMouseDown={[Function]}
  onTouchStart={[Function]}
>
  <div
    data-props={
      Object {
        "item": Object {
          "label": "Item 1",
          "value": "1",
        },
        "removable": false,
        "styles_borderRadius": ".2em",
        "styles_checkRadio_borderWidth": "2px",
        "styles_checkRadio_labelMargin": ".2em",
        "styles_checkRadio_marginBetween": "1em",
        "styles_checkRadio_paddingBottom": ".2em",
        "styles_checkRadio_paddingLeft": ".2em",
        "styles_checkRadio_paddingRight": ".2em",
        "styles_checkRadio_paddingTop": ".2em",
        "styles_colors_background": "white",
        "styles_colors_disabled": "#ECECEC",
        "styles_colors_highlight": "lightblue",
        "styles_colors_primary": "black",
        "styles_colors_secondary": "#C3C3C3",
        "styles_colors_warning": "#FABAAC",
        "styles_colors_warningBold": "#FA2222",
        "styles_disabled": false,
        "styles_fontSize": "1em",
        "styles_hasOptions": true,
        "styles_hasSelection": false,
        "styles_icon_width": "1em",
        "styles_multiple": true,
        "styles_optionHighlighted": Array [],
        "styles_option_paddingBottom": ".25em",
        "styles_option_paddingLeft": ".25em",
        "styles_option_paddingRight": ".25em",
        "styles_option_paddingTop": ".25em",
        "styles_paddingBottom": ".25em",
        "styles_paddingLeft": ".25em",
        "styles_paddingRight": ".25em",
        "styles_paddingTop": ".25em",
        "styles_rightToLeft": false,
        "styles_search_size": "8",
        "styles_selection_margin": ".2em",
        "styles_selection_paddingBottom": ".2em",
        "styles_selection_paddingLeft": ".4em",
        "styles_selection_paddingRight": ".4em",
        "styles_selection_paddingTop": ".2em",
        "styles_width": 600,
        "svg_Remove": undefined,
      }
    }
  />
  <div
    data-props={
      Object {
        "item": Object {
          "label": "Item 2",
          "value": "2",
        },
        "removable": false,
        "styles_borderRadius": ".2em",
        "styles_checkRadio_borderWidth": "2px",
        "styles_checkRadio_labelMargin": ".2em",
        "styles_checkRadio_marginBetween": "1em",
        "styles_checkRadio_paddingBottom": ".2em",
        "styles_checkRadio_paddingLeft": ".2em",
        "styles_checkRadio_paddingRight": ".2em",
        "styles_checkRadio_paddingTop": ".2em",
        "styles_colors_background": "white",
        "styles_colors_disabled": "#ECECEC",
        "styles_colors_highlight": "lightblue",
        "styles_colors_primary": "black",
        "styles_colors_secondary": "#C3C3C3",
        "styles_colors_warning": "#FABAAC",
        "styles_colors_warningBold": "#FA2222",
        "styles_disabled": false,
        "styles_fontSize": "1em",
        "styles_hasOptions": true,
        "styles_hasSelection": false,
        "styles_icon_width": "1em",
        "styles_multiple": true,
        "styles_optionHighlighted": Array [],
        "styles_option_paddingBottom": ".25em",
        "styles_option_paddingLeft": ".25em",
        "styles_option_paddingRight": ".25em",
        "styles_option_paddingTop": ".25em",
        "styles_paddingBottom": ".25em",
        "styles_paddingLeft": ".25em",
        "styles_paddingRight": ".25em",
        "styles_paddingTop": ".25em",
        "styles_rightToLeft": false,
        "styles_search_size": "8",
        "styles_selection_margin": ".2em",
        "styles_selection_paddingBottom": ".2em",
        "styles_selection_paddingLeft": ".4em",
        "styles_selection_paddingRight": ".4em",
        "styles_selection_paddingTop": ".2em",
        "styles_width": 600,
        "svg_Remove": undefined,
      }
    }
  />
  <div
    data-props={
      Object {
        "item": Object {
          "label": "Item 3",
          "value": "3",
        },
        "removable": false,
        "styles_borderRadius": ".2em",
        "styles_checkRadio_borderWidth": "2px",
        "styles_checkRadio_labelMargin": ".2em",
        "styles_checkRadio_marginBetween": "1em",
        "styles_checkRadio_paddingBottom": ".2em",
        "styles_checkRadio_paddingLeft": ".2em",
        "styles_checkRadio_paddingRight": ".2em",
        "styles_checkRadio_paddingTop": ".2em",
        "styles_colors_background": "white",
        "styles_colors_disabled": "#ECECEC",
        "styles_colors_highlight": "lightblue",
        "styles_colors_primary": "black",
        "styles_colors_secondary": "#C3C3C3",
        "styles_colors_warning": "#FABAAC",
        "styles_colors_warningBold": "#FA2222",
        "styles_disabled": false,
        "styles_fontSize": "1em",
        "styles_hasOptions": true,
        "styles_hasSelection": false,
        "styles_icon_width": "1em",
        "styles_multiple": true,
        "styles_optionHighlighted": Array [],
        "styles_option_paddingBottom": ".25em",
        "styles_option_paddingLeft": ".25em",
        "styles_option_paddingRight": ".25em",
        "styles_option_paddingTop": ".25em",
        "styles_paddingBottom": ".25em",
        "styles_paddingLeft": ".25em",
        "styles_paddingRight": ".25em",
        "styles_paddingTop": ".25em",
        "styles_rightToLeft": false,
        "styles_search_size": "8",
        "styles_selection_margin": ".2em",
        "styles_selection_paddingBottom": ".2em",
        "styles_selection_paddingLeft": ".4em",
        "styles_selection_paddingRight": ".4em",
        "styles_selection_paddingTop": ".2em",
        "styles_width": 600,
        "svg_Remove": undefined,
      }
    }
  />
  <div
    data-props={
      Object {
        "item": Object {
          "label": "Item 4",
          "value": "4",
        },
        "removable": false,
        "styles_borderRadius": ".2em",
        "styles_checkRadio_borderWidth": "2px",
        "styles_checkRadio_labelMargin": ".2em",
        "styles_checkRadio_marginBetween": "1em",
        "styles_checkRadio_paddingBottom": ".2em",
        "styles_checkRadio_paddingLeft": ".2em",
        "styles_checkRadio_paddingRight": ".2em",
        "styles_checkRadio_paddingTop": ".2em",
        "styles_colors_background": "white",
        "styles_colors_disabled": "#ECECEC",
        "styles_colors_highlight": "lightblue",
        "styles_colors_primary": "black",
        "styles_colors_secondary": "#C3C3C3",
        "styles_colors_warning": "#FABAAC",
        "styles_colors_warningBold": "#FA2222",
        "styles_disabled": false,
        "styles_fontSize": "1em",
        "styles_hasOptions": true,
        "styles_hasSelection": false,
        "styles_icon_width": "1em",
        "styles_multiple": true,
        "styles_optionHighlighted": Array [],
        "styles_option_paddingBottom": ".25em",
        "styles_option_paddingLeft": ".25em",
        "styles_option_paddingRight": ".25em",
        "styles_option_paddingTop": ".25em",
        "styles_paddingBottom": ".25em",
        "styles_paddingLeft": ".25em",
        "styles_paddingRight": ".25em",
        "styles_paddingTop": ".25em",
        "styles_rightToLeft": false,
        "styles_search_size": "8",
        "styles_selection_margin": ".2em",
        "styles_selection_paddingBottom": ".2em",
        "styles_selection_paddingLeft": ".4em",
        "styles_selection_paddingRight": ".4em",
        "styles_selection_paddingTop": ".2em",
        "styles_width": 600,
        "svg_Remove": undefined,
      }
    }
  />
  <div
    data-props={
      Object {
        "item": Object {
          "label": "Item 5",
          "value": "5",
        },
        "removable": false,
        "styles_borderRadius": ".2em",
        "styles_checkRadio_borderWidth": "2px",
        "styles_checkRadio_labelMargin": ".2em",
        "styles_checkRadio_marginBetween": "1em",
        "styles_checkRadio_paddingBottom": ".2em",
        "styles_checkRadio_paddingLeft": ".2em",
        "styles_checkRadio_paddingRight": ".2em",
        "styles_checkRadio_paddingTop": ".2em",
        "styles_colors_background": "white",
        "styles_colors_disabled": "#ECECEC",
        "styles_colors_highlight": "lightblue",
        "styles_colors_primary": "black",
        "styles_colors_secondary": "#C3C3C3",
        "styles_colors_warning": "#FABAAC",
        "styles_colors_warningBold": "#FA2222",
        "styles_disabled": false,
        "styles_fontSize": "1em",
        "styles_hasOptions": true,
        "styles_hasSelection": false,
        "styles_icon_width": "1em",
        "styles_multiple": true,
        "styles_optionHighlighted": Array [],
        "styles_option_paddingBottom": ".25em",
        "styles_option_paddingLeft": ".25em",
        "styles_option_paddingRight": ".25em",
        "styles_option_paddingTop": ".25em",
        "styles_paddingBottom": ".25em",
        "styles_paddingLeft": ".25em",
        "styles_paddingRight": ".25em",
        "styles_paddingTop": ".25em",
        "styles_rightToLeft": false,
        "styles_search_size": "8",
        "styles_selection_margin": ".2em",
        "styles_selection_paddingBottom": ".2em",
        "styles_selection_paddingLeft": ".4em",
        "styles_selection_paddingRight": ".4em",
        "styles_selection_paddingTop": ".2em",
        "styles_width": 600,
        "svg_Remove": undefined,
      }
    }
  />
</ul>
`;
